src/Blocks/BlockCactus.h

```diff
@@ -23,6 +23,39 @@ class cBlockCactusHandler :
 
 
 
+	/** Called before a cactus block is placed by a player, overrides cItemHandler::GetPlacementBlockTypeMeta().
+	Calls CanBeAt function to determine if a cactus block can be placed on a given block. */
+	bool GetPlacementBlockTypeMeta(
+		cChunkInterface & a_ChunkInterface,
+		cPlayer & a_Player,
+		const Vector3i a_PlacedBlockPos,
+		eBlockFace a_ClickedBlockFace,
+		const Vector3i a_CursorPos,
+		BLOCKTYPE & a_BlockType, NIBBLETYPE & a_BlockMeta
+	) override
+	{
+		if (a_Player.GetWorld()->DoWithChunkAt(a_PlacedBlockPos,
+			[this, a_PlacedBlockPos, &a_ChunkInterface](cChunk & a_Chunk)
+			{
+				auto RelPos = cChunkDef::AbsoluteToRelative(a_PlacedBlockPos);
+				return CanBeAt(a_ChunkInterface, RelPos, a_Chunk);
+			}
+			))
```

Comment:
Style: This should be have one less indent level (so that it's the same level as the `if` the last parenthesis belongs to.)

Reply:
Hopefully my last commits reflected the changes you wanted.